plugins/kubernetes.js

```diff
@@ -0,0 +1,77 @@
+const Client = require("kubernetes-client").Client;
+const aw = require("awaitify-stream");
+
+let kc;
+
+const kubernetes = {
+    name: "Kubernetes",
+    init: (controller) => {
+        controller.addPluginExtension("kubernetes", kubernetes);
+    },
+    getK8s: () => {
+        const version = "1.13";
+        kc = new Client({ version });
+        return kc;
+    },
+    startWatch: (
+        resource,
+        namespace = "default",
+        callback,
+        queryString = null
+    ) => {
+        let run = true;
+        let stream;
+
+        const watcher = async () => {
+            while (run) {
+                let k8s;
+                if (resource.group === "" || resource.version === "") {
+                    k8s = kubernetes.getK8s().api.v1;
+                } else {
+                    k8s = kubernetes.getK8s().apis[resource.group][
+                        resource.version
+                    ];
+                }
+                try {
+                    stream = await k8s.watch
+                        .namespaces(namespace)
+                        [resource.type].getObjectStream(queryString); // eslint-disable-line no-await-in-loop
+                } catch (e) {
+                    console.log(e);
+                }
+                let reader = aw.createReader(stream);
+
+                let object;
+                const startTime = Date.now();
+                try {
+                    while (null !== (object = await reader.readAsync())) {
+                        const newEventTime = Date.parse(
+                            object.object.metadata.creationTimestamp
+                        );
+                        if (newEventTime > startTime) {
+                            callback(object.type, object.object);
+                        }
+                        if (run === false) {
+                            stream.destroy();
+                            break;
+                        }
+                    }
+                } catch (e) {
+                    console.log(e);
+                }
+            }
+        };
+        const result = watcher();
+
+        return async () => {
```

Comment:
doesn't look like this anonymous function uses the `await` keyword, so `async` is not needed here.

Reply:
Yeah definitely not needed. I think it makes the code easier to understand since the function is intended to be used in an asynchronous way so you can block until the watch function exits. Does it hurt anything defining it as async or it is just unnecessary? I don't really feel strongly about this but just wondering out loud if explicitly defining a function that returns a promise as async makes the code more readable?

Reply:
I would actually argue that purposely using a reserved keyword in a language for any reason other than its intended use case makes the code more difficult to understand, not less.

also, maybe I'm missing something here, but this anonymous function doesn't appear to block at all.  so using `async` here is misleading in more than one way - in the sense that it's only needed to facilitate `await`, and that the function isn't asynchronous.

Reply:
to answer your other question, unnecessary usage of `async` probably doesn't have any consequences besides what I mentioned about the code being more difficult to understand.

Reply:
I've gone ahead and removed the use of async here. While I agree with you @ChrisSchreiber that it _can_ point to the user that this function needs to be blocked until watcher stops, I agree with @mrparkers more that using a reserved keyword without needing to can lead to confusion.